examples/physics/basic/example.js

```diff
@@ -0,0 +1,265 @@
+  ball.checkBoundary(rightWall);
+
+  // displays the ball and the ground/walls 
+  ball.render();
+  ground.render();
+  leftWall.render();
+  rightWall.render();
+}
+
+class Ball {
+  // create a class for the bouncing ball
+  // the ball has it's own size, color, and x, y, z positions
+  // as well as it's own physics forces applied to it
+  constructor(x, y, z) {
+    this.position = createVector(x, y, z);
+    this.startingPosition = createVector(x, y, z);
+    this.velocity = createVector(0, .01, 0);
+    this.acceleration = createVector(0, 0, 0);
+    this.held = false;
+    this.radius = 0.25;
+    this.color = color(50, 50, 200);
+    this.heldColor = color(200, 50, 50);
+  }
+
+  // applies acceleration force to the ball
+  applyForce(force) {
+    this.acceleration.add(force);
+  }
+
+  update() {
+    if (this.position.y < ground.position.y) {
+      // if the y-position of the ball is less than the y-position of the ground
+      // reset the ball to it's starting position
+      this.reset();
+      return;
+    }
+    if (this.held) {
+      // if the ball is held, set it's velocity to 0 to stop it from moving
+      this.velocity.set(0, 0, 0);
+    } else {
+      // otherwise, if the ball isn't being held 
+      // add acceleration and velocity to the ball so that it moves
+      this.velocity.add(this.acceleration);
+      this.position.add(this.velocity);
+    }
+    this.acceleration.set(0, 0, 0);
+  }
+
+  // if the ball escapes the boundaries for some reason
+  // this resets the ball's current position to its starting position
+  reset() {
+    this.position.set(this.startingPosition);
+    this.velocity.set(0, 0.01, 0);
+    this.acceleration.set(0, 0, 0);
+  }
+
+  // checks for the intersection of the ball's x, y, z positions with the x, y, z positions of the boundaries
+  checkIntersection(position, radius) {
+    let d = dist(position.x, position.y, position.z, this.position.x, this.position.y, this.position.z);
+    return (d <= radius + this.radius)
+  }
+
+
+  // if the ball has intersected with a boundary, check the dot() product after a collision
+  // a dot product is a way of asking whether two vectors are facing the same direction
+  // in this case, if it's less 0, then no. otherwise, yes. 
+  checkBoundary(boundary) {
+    if (boundary.checkIntersection(this.position, this.radius)) {
+      if (p5.Vector.dot(boundary.reflectionNormal, this.velocity) <= 0) {
+        return;
+      }
+      // velocity of the ball is reversed upon collision
+      // this causes the ball to bounce back up
+      this.velocity.mult(boundary.reflectionNormal);
+
+      // change the color of the sphere everytime it hits the ground
+      this.color = color(random(255), random(255), random(255));
+
+    }
+  }
+
+  render() {
+    push();
+    // if the ball is held...
+    // make it a different color than normal
+    if (this.held) {
+      fill(this.heldColor);
+    } else {
+      fill(this.color);
```

Comment:
replace fill with emmisiveMaterial instead of doing both so this can read:
```js
 if (this.held) {
      emissiveMaterial(this.heldColor);
    } else {
      emissiveMaterial(this.color);
  }
```
and you can get rid of the emmisiveMaterial call below. This just makes it so that the ball is not influenced by the lighting.